service/src/main/java/com/commercetools/pspadapter/payone/domain/payone/model/common/Notification.java

```diff
@@ -58,13 +63,30 @@
 
     public static Notification fromKeyValueString(final String keyValueString, final String separatorPattern) {
 
-        Map<String, String> notificationValues = Splitter.onPattern(separatorPattern).withKeyValueSeparator("=").split(keyValueString);
+        final Map<String, String> notificationValues = Pattern.compile(separatorPattern)
+               .splitAsStream(keyValueString.trim())
+               .map(s -> s.split("="))
+               .collect(Collectors.toMap(a -> a[0], a -> a.length>1? a[1]: ""));
```

Comment:
I don't understand this part. Can you add a comment to the code what it means?

Reply:
Also, why >1 and not >0? Can `a` be null?

Reply:
@lojzatran 01bd736

Reply:
@butenkor the `s.split("=")` method will return an Array - which may be empty but never null; It could have only one entry (e.g. if substring is "key="), thats why its checking for `length > 1` and just put the value if its there.
I'm curious what happens if `a[0]` is empty string... i'll check that.

Reply:
a366697